### ISSUE TITLE
What happened to displacy?

### ISSUE THREAD

**Issue by nathanathan** [spacy-io/spaCy#426]
The displacy page at spacy.io/demos/displacy now returns a 404. It was a very useful reference for working with dependency trees in Spacy. I hope it will be coming back.

**Comment by lx5491** [comment]
I don't know either. That helps me with visualization. I sent them an email but got no reply till now

**Comment by arvindsg** [comment]
Can we get the source for displacy, was  great to have the same available as pretty webservice for quick visualization

**Comment by aliabbasjp** [comment]
Mee too

**Comment by sderygithub** [comment]
@nathanathan @lx5491 @arvindsg @aliabbasjp Facing the same issue :( so decided to write an alternative for until the full Displacy comes back. At the moment it's a simple Flask application (https://github.com/sderygithub/NLTree) you'll have to run locally.. I may just push it on my server in a few days and host it. You can see on my website what it currently looks like (http://dery.xyz/projects/nltree/overview.html)

**Comment by syllog1sm** [comment]
I hope it will be back soon too. I've been asking why it was taken offline as well.

**Comment by honnibal** [comment]
The displaCy and sense2vec demos should now be back online: https://spacy.io/demos

They're both running on the same server, and there may be some short term instability. Fingers crossed! The demos aren't hugely resource intensive, so in theory everything should be fine; but it's still a slightly rickety arrangement.

The good news is that both demos will be open-sourced. I hope to flip the repos to public tonight. Documentation will be a bit sparse, but they should be reasonably easy to host. If anyone is able to donate some compute resources, please get in touch: honnibal@gmail.com.

Unfortunately I'm away for the next week (timing...), but I'll do my best to keep things up. After I get back, the updates should be coming quite quickly.

**Comment by honnibal** [comment]
The displaCy server code is now open-source: https://github.com/spacy-io/displacy-server :tada:

Hopefully, it will be easy for someone else to put up a mirror if my server dies while I'm away. I've also added @ines to the spacy-io Github organisation, so that she can clean up the sense2vec demo for release once she has time.

**Comment by sderygithub** [comment]
@honnibal Brilliant!

**Comment by ines** [comment]
Yay, so nice to see displaCy back online! 🎉  And thanks for adding me back to the GitHub org – I'll try my best to help out with everything demo-related while @honnibal is away.

A little note on the displaCy front-end code: I developed the core of this over a year ago with a pretty specific (and now fairly outdated) [concept](https://ines.io/blog/developing-displacy) in mind. It also hasn't been maintained for months, so there's definitely room for improvement.

I've already started re-writing the basic visualiser to switch it over to SVGs and use a more logical JSON structure to describe the words, arcs and relations (because to be honest, the current one is kinda messy). Using SVGs will also make easier to export the graphics and embed them in documents and blog posts. At the moment, I'm thinking about moving more towards a cleaner and more reliable visualiser and not focusing too much on the rather abstract manual annotation mode. 

But I'm curious to hear your suggestions and feature requests!

Tagging everyone who participated here just in case, since this issue is kinda old.
/cc @nathanathan, @lx5491, @arvindsg, @aliabbasjp, @sderygithub
